# Characterizing the Non-Coding Genome of Glial Cells in the Developing Human Cortex
 
#### ExtFig1: Quality check of multi-omic datasets #### 
    ExtFig1a: RNA DeSEQ2 Correlation
    ExtFig1b: ATAC-seq count Pearson Correlation
    ExtFig1c: WGBS Correlation
    ExtFig1d: PLAC-seq HPrep Heatmap
    ExtFig1e: Cibersort prediction on bulk RNA-seq datasets
    
#### ExtFig2: Quality check of multi-omic datasets #### 
    ExtFig2a: Barplot of how cCREs are defined
    ExtFig2b: Upset plot of H3K4me3 anchors
    ExtFig2c: Barplot of LMAR overlapping H3K4me3 anchors
    ExtFig2d: Barplot of number of interactions per an anchor bin 
    ExtFig2e: Cumulative Distribution of interaction distance for all 4 cell types
    ExtFig2f: Overview of interaction clusters
    ExtFig2g: Violin plot of observed over expected counts for LMAR interaction vs other interactions
    ExtFig2h: Barplot showing the percentage of clusters containing a LMAR
    

#### ExtFig3: Vista enhancers involved in the 3D epigenome of the developing brain. #### 
    ExtFig3a: Upset plot of significant interactions
    ExtFig3b: Heatmap of correlation between RPKM and Interaction difference for all distal interacitons and distal interactions with LMAR in the distal bin.
    ExtFig3f: Linking Neu Vista to their potential target genes.
    
#### ExtFig4: #### 

#### ExtFig5: #### 

#### ExtFig6: #### 

#### ExtFig7: #### 

#### ExtFig8: #### 