# Characterizing the Non-Coding Genome of Glial Cells in the Developing Human Cortex
 
### ExtFig1: Quality check of multi-omic datasets
    ExtFig1a: Hierarhcial cluster of transcriptome created with DeSEQ2
    ExtFig1b: Spearman correlation of ATAC-seq reads within unified peakset
    ExtFig1c: CpG clustering
    ExtFig1d: Pearson correlation of PLAC-seq libraries created with HPrep
    ExtFig1e: Civersort prediction of cell distribution imputed from Nowakaski et al. 2016
    
### ExtFig2: Quality check of multi-omic datasets
    ExtFig2a: Barplot of how cCREs are defined
    ExtFig2b: Upset plot of H3K4me3 anchors
    ExtFig2c: Barplot of LMAR overlapping H3K4me3 anchors
    ExtFig2d: Barplot of number of interactions per an anchor bin 
    ExtFig2e: Cumulative Distribution of interaction distance for all 4 cell types
    ExtFig2f: Overview of interaction clusters
    ExtFig2g: Violin plot of observed over expected counts for LMAR interaction vs other interactions
    ExtFig2h: Barplot showing the percentage of clusters containing a LMAR
    

### ExtFig3: Vista enhancers involved in the 3D epigenome of the developing brain. 
    ExtFig3a: Upset plot of significant interactions
    ExtFig3b: Heatmap of correlation between RPKM and Interaction difference for all distal interacitons and distal interactions with LMAR in the distal bin.
    ExtFig3f: Linking Neu Vista to their potential target genes.
    
### ExtFig4: 

### ExtFig5:

### ExtFig6:

### ExtFig7: 

### ExtFig8: